src/main/java/org/opensearch/ad/util/IndexUtils.java

```diff
@@ -113,9 +116,9 @@ public IndexUtils(
      * @throws IllegalArgumentException Thrown when an alias is passed in that points to more than one index
      */
     public String getIndexHealthStatus(String indexOrAliasName) throws IllegalArgumentException {
-        if (!clusterService.state().getRoutingTable().hasIndex(indexOrAliasName)) {
+        if (!indexExists(indexOrAliasName)) {
             // Check if the index is actually an alias
-            if (clusterService.state().metadata().hasAlias(indexOrAliasName)) {
+            if (aliasExists(indexOrAliasName)) {
```

Comment:
These two calls *might* be able to be combined; instead of checking existence, you could "get" the index and then something in the index metadata indicates whether it's an alias.... I think.   Not for this PR but maybe consider that a TODO.